src/hooks/use-debounce.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from "react";
+
+export const useDebounce = (value: string, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
```

Comment:
훅으로 분리 잘 하셨네요 👍🏽 

```suggestion
export const useDebounce = <T>(value: T, delay: number): T => {
  const [debouncedValue, setDebouncedValue] = useState<T>(value);
```

다양한 곳에서 쓰기 위해 value 를 string 타입으로 고정하지 않고 이렇게 제네릭으로 처리할 수도 있습니다!